flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/NetworkEnvironmentConfiguration.java

```diff
@@ -105,16 +112,381 @@ public NettyConfig nettyConfig() {
 		return nettyConfig;
 	}
 
+	public boolean isCreditBased() {
+		return isCreditBased;
+	}
+
+	// ------------------------------------------------------------------------
+
+	/**
+	 * Utility method to extract network related parameters from the configuration and to
+	 * sanity check them.
+	 *
+	 * @param configuration configuration object
+	 * @param maxJvmHeapMemory the maximum JVM heap size (in bytes)
+	 * @param localTaskManagerCommunication true, to skip initializing the network stack
+	 * @param taskManagerAddress identifying the IP address under which the TaskManager will be accessible
+	 * @return NetworkEnvironmentConfiguration
+	 */
+	@Deprecated
+	public static NetworkEnvironmentConfiguration fromConfiguration(
+		Configuration configuration,
+		long maxJvmHeapMemory,
+		boolean localTaskManagerCommunication,
+		InetAddress taskManagerAddress) {
+
+		final int dataport = getDataport(configuration);
+
+		final int pageSize = getPageSize(configuration);
+
+		final int numberOfNetworkBuffers = calculateNumberOfNetworkBuffers(configuration, maxJvmHeapMemory);
+
+		final NettyConfig nettyConfig = createNettyConfig(configuration, localTaskManagerCommunication, taskManagerAddress, dataport);
+
+		int initialRequestBackoff = configuration.getInteger(TaskManagerOptions.NETWORK_REQUEST_BACKOFF_INITIAL);
```

Comment:
We should also try to move all options which are used only for network environment from `TaskManagerOptions` to new class `NetworkEnvironmentOptions` to separate them for different shuffle services. What do you think?

Reply:
That is a very good idea. I would create a separate jira for this task later.

Reply:
Agreed, let's create an issue